### PR TITLE
python3Packages.keras: 3.10.0 -> 3.11.1

### PR DESCRIPTION
**Pull request by GaetanLepage**
## Things done

Diff: https://github.com/keras-team/keras/compare/refs/tags/v3.10.0...refs/tags/v3.11.0
Changelog: https://github.com/keras-team/keras/releases/tag/v3.11.0

cc 

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 429693`
Commit: `73b219e4b8bfbf434b2b5c49b63cd58c69a398b5`

---
### `x86_64-linux`
<details>
  <summary>:x: 50 packages failed to build:</summary>
  <ul>
    <li>python312Packages.ale-py</li>
    <li>python312Packages.ale-py.dist</li>
    <li>python312Packages.arviz</li>
    <li>python312Packages.arviz.dist</li>
    <li>python312Packages.bambi</li>
    <li>python312Packages.bambi.dist</li>
    <li>python312Packages.brax</li>
    <li>python312Packages.brax.dist</li>
    <li>python312Packages.corner</li>
    <li>python312Packages.corner.dist</li>
    <li>python312Packages.dalle-mini</li>
    <li>python312Packages.dalle-mini.dist</li>
    <li>python312Packages.evosax</li>
    <li>python312Packages.evosax.dist</li>
    <li>python312Packages.flax</li>
    <li>python312Packages.flax.dist</li>
    <li>python312Packages.flowjax</li>
    <li>python312Packages.flowjax.dist</li>
    <li>python312Packages.gymnasium</li>
    <li>python312Packages.gymnasium.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.numpyro</li>
    <li>python312Packages.numpyro.dist</li>
    <li>python312Packages.nutpie</li>
    <li>python312Packages.nutpie.dist</li>
    <li>python312Packages.pettingzoo</li>
    <li>python312Packages.pettingzoo.dist</li>
    <li>python312Packages.pymc</li>
    <li>python312Packages.pymc.dist</li>
    <li>python312Packages.shimmy</li>
    <li>python312Packages.shimmy.dist</li>
    <li>python312Packages.skrl</li>
    <li>python312Packages.skrl.dist</li>
    <li>python312Packages.stable-baselines3</li>
    <li>python312Packages.stable-baselines3.dist</li>
    <li>python312Packages.tianshou</li>
    <li>python312Packages.tianshou.dist</li>
    <li>python312Packages.torchrl</li>
    <li>python312Packages.torchrl.dist</li>
    <li>python312Packages.tyro</li>
    <li>python312Packages.tyro.dist</li>
    <li>python312Packages.unsloth</li>
    <li>python312Packages.unsloth-zoo</li>
    <li>python312Packages.unsloth-zoo.dist</li>
    <li>python312Packages.unsloth.dist</li>
    <li>python312Packages.vqgan-jax</li>
    <li>python312Packages.vqgan-jax.dist</li>
    <li>python312Packages.waymax</li>
    <li>python312Packages.waymax.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 24 packages built:</summary>
  <ul>
    <li>python312Packages.cleanlab</li>
    <li>python312Packages.cleanlab.dist</li>
    <li>python312Packages.deepface</li>
    <li>python312Packages.deepface.dist</li>
    <li>python312Packages.distrax</li>
    <li>python312Packages.distrax.dist</li>
    <li>python312Packages.dm-sonnet</li>
    <li>python312Packages.dm-sonnet.dist</li>
    <li>python312Packages.edward</li>
    <li>python312Packages.edward.dist</li>
    <li>python312Packages.keras</li>
    <li>python312Packages.keras.dist</li>
    <li>python312Packages.mhcflurry</li>
    <li>python312Packages.mhcflurry.dist</li>
    <li>python312Packages.mtcnn</li>
    <li>python312Packages.mtcnn.dist</li>
    <li>python312Packages.pytensor</li>
    <li>python312Packages.pytensor.dist</li>
    <li>python312Packages.retinaface</li>
    <li>python312Packages.retinaface.dist</li>
    <li>python312Packages.rlax</li>
    <li>python312Packages.rlax.dist</li>
    <li>python312Packages.tensorflow-probability</li>
    <li>python312Packages.tensorflow-probability.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python312Packages.flax</summary>
<pre>  File "/nix/store/m32f8511ibiaw139nf7pw97rlryzyw2x-python3.12-execnet-2.1.1/lib/python3.12/site-packages/execnet/gateway_base.py", line 1273 in serve
  File "/nix/store/m32f8511ibiaw139nf7pw97rlryzyw2x-python3.12-execnet-2.1.1/lib/python3.12/site-packages/execnet/gateway_base.py", line 1806 in serve
  File "<string>", line 8 in <module>
  File "<string>", line 1 in <module>

Extension modules: jaxlib.cpu_feature_guard, numpy._core._multiarray_umath, numpy.linalg._umath_linalg, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._pcg64, numpy.random._mt19937, numpy.random._generator, numpy.random._philox, numpy.random._sfc64, numpy.random.mtrand, .simplejson._speedups, google._upb._message, msgpack._cmsgpack, yaml._yaml, charset_normalizer.md, requests.packages.charset_normalizer.md, requests.packages.chardet.md, h5py._errors, h5py.defs, h5py._objects, h5py.h5, h5py.utils, h5py.h5t, h5py.h5s, h5py.h5ac, h5py.h5p, h5py.h5r, h5py._proxy, h5py._conv, h5py.h5z, h5py.h5a, h5py.h5d, h5py.h5ds, h5py.h5g, h5py.h5i, h5py.h5o, h5py.h5f, h5py.h5fd, h5py.h5pl, h5py.h5l, h5py._selector, scipy._lib._ccallback_c, scipy.sparse._sparsetools, _csparsetools, scipy.sparse._csparsetools, psutil._psutil_linux, psutil._psutil_posix, sklearn.__check_build._check_build, lz4._version, lz4.frame._frame, scipy.special._ufuncs_cxx, scipy.special._ellip_harm_2, scipy.special._special_ufuncs, scipy.special._gufuncs, scipy.special._ufuncs, scipy.special._specfun, scipy.special._comb, scipy.linalg._fblas, scipy.linalg._flapack, scipy.linalg.cython_lapack, scipy.linalg._cythonized_array_utils, scipy.linalg._solve_toeplitz, scipy.linalg._decomp_lu_cython, scipy.linalg._matfuncs_schur_sqrtm, scipy.linalg._matfuncs_expm, scipy.linalg._linalg_pythran, scipy.linalg.cython_blas, scipy.linalg._decomp_update, scipy.sparse.linalg._dsolve._superlu, scipy.sparse.linalg._eigen.arpack._arpack, scipy.sparse.linalg._propack._spropack, scipy.sparse.linalg._propack._dpropack, scipy.sparse.linalg._propack._cpropack, scipy.sparse.linalg._propack._zpropack, scipy.spatial._ckdtree, scipy._lib.messagestream, scipy.spatial._qhull, scipy.spatial._voronoi, scipy.spatial._hausdorff, scipy.spatial._distance_wrap, scipy.spatial.transform._rotation, scipy.spatial.transform._rigid_transform, scipy.optimize._group_columns, scipy.optimize._trlib._trlib, scipy.optimize._lbfgsb, _moduleTNC, scipy.optimize._moduleTNC, scipy.optimize._slsqplib, scipy.optimize._minpack, scipy.optimize._lsq.givens_elimination, scipy.optimize._zeros, scipy._lib._uarray._uarray, scipy.linalg._decomp_interpolative, scipy.optimize._bglu_dense, scipy.optimize._lsap, scipy.optimize._direct, scipy.integrate._odepack, scipy.integrate._quadpack, scipy.integrate._vode, scipy.integrate._dop, scipy.integrate._lsoda, scipy.interpolate._fitpack, scipy.interpolate._dfitpack, scipy.interpolate._dierckx, scipy.interpolate._ppoly, scipy.interpolate._interpnd, scipy.interpolate._rbfinterp_pythran, scipy.interpolate._rgi_cython, scipy.special.cython_special, scipy.stats._stats, scipy.stats._biasedurn, scipy.stats._stats_pythran, scipy.stats._levy_stable.levyst, scipy.stats._ansari_swilk_statistics, scipy.sparse.csgraph._tools, scipy.sparse.csgraph._shortest_path, scipy.sparse.csgraph._traversal, scipy.sparse.csgraph._min_spanning_tree, scipy.sparse.csgraph._flow, scipy.sparse.csgraph._matching, scipy.sparse.csgraph._reordering, scipy.stats._sobol, scipy.stats._qmc_cy, scipy.stats._rcont.rcont, scipy.stats._qmvnt_cy, scipy.ndimage._nd_image, scipy.ndimage._rank_filter_1d, _ni_label, scipy.ndimage._ni_label, sklearn.utils._isfinite, sklearn.utils.sparsefuncs_fast, sklearn.utils.murmurhash, sklearn.utils._openmp_helpers (total: 135)
....................... [ 82%]
.........[gw10] node down: Not properly terminated
F
replacing crashed worker gw10

collecting: 32/33 workers.......................................s....................
collecting: 32/33 workers.............................................................................................................................................................................................................................................................................................                                                                          [100%]
=================================== FAILURES ===================================
______________________ tests/nnx/nn/normalization_test.py ______________________
[gw10] linux -- Python 3.12.11 /nix/store/2i76abyx06asxr2k3rgsp8zx647qj38h-python3-3.12.11/bin/python3.12
worker 'gw10' crashed while running 'tests/nnx/nn/normalization_test.py::TestLinenConsistency::test_nnx_linen_groupnorm_equivalence0'
=========================== short test summary info ============================
FAILED tests/nnx/nn/normalization_test.py::TestLinenConsistency::test_nnx_linen_groupnorm_equivalence0
================= 1 failed, 2056 passed, 26 skipped in 41.56s ==================</pre>
</details>
</details>

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 429693`
Commit: `73b219e4b8bfbf434b2b5c49b63cd58c69a398b5`

---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python312Packages.torchrl</li>
    <li>python312Packages.torchrl.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 72 packages built:</summary>
  <ul>
    <li>python312Packages.ale-py</li>
    <li>python312Packages.ale-py.dist</li>
    <li>python312Packages.arviz</li>
    <li>python312Packages.arviz.dist</li>
    <li>python312Packages.bambi</li>
    <li>python312Packages.bambi.dist</li>
    <li>python312Packages.brax</li>
    <li>python312Packages.brax.dist</li>
    <li>python312Packages.cleanlab</li>
    <li>python312Packages.cleanlab.dist</li>
    <li>python312Packages.corner</li>
    <li>python312Packages.corner.dist</li>
    <li>python312Packages.dalle-mini</li>
    <li>python312Packages.dalle-mini.dist</li>
    <li>python312Packages.deepface</li>
    <li>python312Packages.deepface.dist</li>
    <li>python312Packages.distrax</li>
    <li>python312Packages.distrax.dist</li>
    <li>python312Packages.dm-sonnet</li>
    <li>python312Packages.dm-sonnet.dist</li>
    <li>python312Packages.edward</li>
    <li>python312Packages.edward.dist</li>
    <li>python312Packages.evosax</li>
    <li>python312Packages.evosax.dist</li>
    <li>python312Packages.flax</li>
    <li>python312Packages.flax.dist</li>
    <li>python312Packages.flowjax</li>
    <li>python312Packages.flowjax.dist</li>
    <li>python312Packages.gymnasium</li>
    <li>python312Packages.gymnasium.dist</li>
    <li>python312Packages.keras</li>
    <li>python312Packages.keras.dist</li>
    <li>python312Packages.mhcflurry</li>
    <li>python312Packages.mhcflurry.dist</li>
    <li>python312Packages.minari</li>
    <li>python312Packages.minari.dist</li>
    <li>python312Packages.mtcnn</li>
    <li>python312Packages.mtcnn.dist</li>
    <li>python312Packages.numpyro</li>
    <li>python312Packages.numpyro.dist</li>
    <li>python312Packages.nutpie</li>
    <li>python312Packages.nutpie.dist</li>
    <li>python312Packages.pettingzoo</li>
    <li>python312Packages.pettingzoo.dist</li>
    <li>python312Packages.pymc</li>
    <li>python312Packages.pymc.dist</li>
    <li>python312Packages.pytensor</li>
    <li>python312Packages.pytensor.dist</li>
    <li>python312Packages.retinaface</li>
    <li>python312Packages.retinaface.dist</li>
    <li>python312Packages.rlax</li>
    <li>python312Packages.rlax.dist</li>
    <li>python312Packages.shimmy</li>
    <li>python312Packages.shimmy.dist</li>
    <li>python312Packages.skrl</li>
    <li>python312Packages.skrl.dist</li>
    <li>python312Packages.stable-baselines3</li>
    <li>python312Packages.stable-baselines3.dist</li>
    <li>python312Packages.tensorflow-probability</li>
    <li>python312Packages.tensorflow-probability.dist</li>
    <li>python312Packages.tianshou</li>
    <li>python312Packages.tianshou.dist</li>
    <li>python312Packages.tyro</li>
    <li>python312Packages.tyro.dist</li>
    <li>python312Packages.unsloth</li>
    <li>python312Packages.unsloth-zoo</li>
    <li>python312Packages.unsloth-zoo.dist</li>
    <li>python312Packages.unsloth.dist</li>
    <li>python312Packages.vqgan-jax</li>
    <li>python312Packages.vqgan-jax.dist</li>
    <li>python312Packages.waymax</li>
    <li>python312Packages.waymax.dist</li>
  </ul>
</details>

**Review by kirillrdy — APPROVED** [review]
